File: 1665-minimum-initial-energy-to-finish-tasks/1665-minimum-initial-energy-to-finish-tasks.py
class Solution(object):
    def minimumEffort(self, tasks):
        """
        :type tasks: List[List[int]]
        :rtype: int
        """
        tasks.sort(key=lambda x: x[1] - x[0])
        ret = 0
        for t in tasks:
            ret = max(ret + t[0], t[1])
        return ret

File: 1665-minimum-initial-energy-to-finish-tasks/1665-minimum-initial-energy-to-finish-tasks.cpp
class Solution {
public:
    static bool cmp(vector<int>& a, vector<int>& b) {
        return a[1] - a[0] < b[1] - b[0];
    }
    int minimumEffort(vector<vector<int>>& tasks) {
        sort(tasks.begin(), tasks.end(), cmp);
        int ret = 0;
        for(auto t : tasks) ret = max(ret + t[0], t[1]);
        return ret;
    }
};